tools/lsp/preview.rs

```diff
@@ -219,17 +222,85 @@ struct PreviewState {
 }
 thread_local! {static PREVIEW_STATE: std::cell::RefCell<PreviewState> = Default::default();}
 
+pub fn set_design_mode(sender: crate::ServerNotifier, enable: bool) {
+    let mut cache = CONTENT_CACHE.get_or_init(Default::default).lock().unwrap();
+    cache.design_mode = enable;
+
+    configure_design_mode(enable, &sender);
+    send_notification(
+        &sender,
+        if enable { "Design mode enabled." } else { "Design mode disabled." },
+        Health::Ok,
+    );
+}
+
+fn show_document_request_from_element_callback(
+    url: &str,
+    start_line: u32,
+    start_column: u32,
+    end_line: u32,
+    end_column: u32,
```

Comment:
This could be a converted to a Range before. So many arguments are quite confusing and easy to get wrong